app/browser/api/ledgerNotifications.js

```diff
@@ -307,8 +342,48 @@ const showBraveWalletUpdated = () => {
   })
 }
 
+const onPromotionReceived = (state) => {
+  const promotion = ledgerState.getPromotionNotification(state)
+
+  if (!promotion.isEmpty() && !promotion.has('firstShowTimestamp')) {
+    state = ledgerState.setPromotionNotificationProp(state, 'firstShowTimestamp', new Date().getTime())
+    showPromotionNotification(state)
+  }
+
+  return state
+}
+
+const showPromotionNotification = (state) => {
+  const notification = ledgerState.getPromotionNotification(state)
+
+  if (notification.isEmpty()) {
+    return
+  }
+
+  appActions.showNotification(notification.toJS())
+}
+
+const removePromotionNotification = (state) => {
+  const notification = ledgerState.getPromotionNotification(state)
+
+  if (notification.isEmpty()) {
+    return
+  }
+
+  appActions.hideNotification(notification.get('message'))
+}
+
 if (ipc) {
-  ipc.on(messages.NOTIFICATION_RESPONSE, (e, message, buttonIndex) => {
+  ipc.on(messages.NOTIFICATION_RESPONSE, (e, message, buttonIndex, checkbox, index, buttonActionId) => {
```

Comment:
is this supposed to do something with `checkbox` and `index`?

Reply:
no we don't need this two params, but we need the last one `buttonActionId`, so I needed to add them